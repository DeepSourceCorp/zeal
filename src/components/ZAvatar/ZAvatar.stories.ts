import ZAvatar from './ZAvatar.vue'

export default {
  title: 'Avatar',
  component: ZAvatar,
  excludeStories: /.*Data$/
}

export const AsALink = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        type="link"
        image="https://randomuser.me/api/portraits/women/24.jpg"
        href="https://example.com"
      ></z-avatar>
    </div>
    `
})

export const InLoadingState = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        :loading="true"
        type="link"
        image="https://randomuser.me/api/portraits/women/24.jpg"
        href="https://example.com"
      ></z-avatar>
    </div>
    `
})


export const WithUserName = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        user-name="Akshay Paliwal"
      ></z-avatar>
    </div>
    `
})

export const WithImageAndUserName = () => ({
  components: { ZAvatar },
  template: `<div class='padded-container'>
      <z-avatar
        image="https://randomuser.me/api/portraits/women/24.jpg"
      ></z-avatar>
    </div>`
})

export const WithSizes = () => ({
  components: { ZAvatar },
  template: `
    <div class='bg-dark padded-container'>
      <div class='padded-container'>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xs"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="sm"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="md"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="lg"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xl"
        ></z-avatar>
      </div>
      <div class='padded-container bg-vanilla-100'>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xs"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="sm"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="md"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="lg"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xl"
        ></z-avatar>
      </div>
      <div class='padded-container'>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xs"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="md"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="lg"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xl"
        ></z-avatar>
      </div>
      <div class='bg-vanilla-100 padded-container'>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xs"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="md"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="lg"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xl"
        ></z-avatar>
      </div>
    </div>
    `
})
