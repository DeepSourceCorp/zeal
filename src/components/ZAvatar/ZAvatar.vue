<template>
  <component v-if="isVisible" :is="TYPES[type]" class="inline-block rounded-full" :class="stroke">
    <span
      v-if="isLoading"
      :class="[`${SIZES[getSize].classes}`]"
      class="flex items-center justify-center text-center rounded-full bg-gradient-skeleton text-vanilla-300"
    >
    </span>
    <template v-else>
      <img
        v-if="this.image"
        :src="image"
        :alt="userName"
        class="rounded-full"
        :class="[`${SIZES[getSize].classes}`]"
        @error.once="setFallbackImage"
      />
      <span
        v-if="!this.image && this.userName"
        :class="[`${SIZES[getSize].classes}`]"
        class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300"
      >
        {{ getUserInitials }}
      </span>
    </template>
  </component>
</template>

<script lang="ts">
import Vue from 'vue'
const SIZES = {
  '2xs': { classes: 'leading-none h-3 w-3 text-xxs', text: '2xs' },
  xs: { classes: 'leading-none h-4 w-4 text-xxs', text: 'xs' },
  sm: { classes: 'leading-none h-6 w-6 text-xs', text: 'sm' },
  md: { classes: 'leading-none h-8 w-8 text-sm', text: 'md' },
  lg: { classes: 'leading-none h-12 w-12 text-lg', text: 'lg' },
  xl: { classes: 'leading-none h-16 w-16 text-2xl', text: 'xl' }
}

const TYPES = {
  link: 'a',
  div: 'div',
  span: 'span'
}

export default Vue.extend({
  name: 'ZAvatar',
  props: {
    type: {
      type: String,
      default: 'link',
      validator: (type) => Object.keys(TYPES).includes(type)
    },
    loading: {
      type: Boolean,
      default: false
    },
    image: String,
    fallbackImage: {
      type: String,
      required: false,
      default: ''
    },
    userName: {
      type: String,
      required: true
    },
    size: {
      type: String,
      default: SIZES.md.text,
      validator: (size) => Object.keys(SIZES).includes(size)
    },
    stroke: {
      type: String,
      required: false,
      default: 'bg-ink-300 p-0.5'
    }
  },
  data() {
    return {
      isVisible: true,
      SIZES,
      TYPES
    }
  },
  computed: {
    isLoading(): boolean {
      const { $options, $props } = this.$parent as Vue
      return $options.name === 'ZAvatarGroup' ? $props.loading : this.loading
    },
    getSize(): string {
      /**
       * Returns `size` property as size of avatar if parent is not ZAvatarGroup,
       * else it returns `size` property of parent as size of avatar.
       */
      const { $options, $props } = this.$parent as Vue
      return $options.name === 'ZAvatarGroup' ? $props.size : this.size
    },
    getUserInitials(): string {
      /**
       * Returns capitalized initials of the user.
       * (Only from the first two words of the userName)
       */
      return this.userName
        ? this.userName
            .split(' ')
            .map((elem) => elem.charAt(0))
            .slice(0, this.size === '2xs' ? 1 : 2) // Make it a single alphabet for the `2xs` size variant
            .join('')
            .toUpperCase()
        : ''
    }
  },
  methods: {
    setFallbackImage(event: ErrorEvent): void {
      if (!this.fallbackImage) return

      const el = event.target as HTMLImageElement
      if (el.src !== this.fallbackImage) {
        el.src = this.fallbackImage
      }
    }
  }
})
</script>
