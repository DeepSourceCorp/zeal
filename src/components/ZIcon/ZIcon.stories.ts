import ZIcon from './ZIcon.vue'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Icons',
  component: ZIcon,
  excludeStories: /.*Data$/
}

export const Default = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up'></z-icon>
        <p class="icon-title">arrow-up</p>
    </div>`
})

export const IconWithCustomColor = () => ({
  components: { ZIcon },
  template: `<div class="wrapper">
        <z-icon icon="activity" color="juniper"></z-icon>
        <p class="icon-title">activity</p>
    </div>`
})

export const SmallIcon = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up' size="small"></z-icon>
        <p class='icon-title'>arrow-up</p>
    </div>`
})
