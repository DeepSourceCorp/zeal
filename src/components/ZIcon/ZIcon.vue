<template>
    <svg 
      :class="[
        'z-icon',
        `${size && `z-icon--${size}`}`,
        `${icon && `z-icon--${icon}`}`
      ]"
      viewBox="0 0 24 24" 
      v-html="getIcon(icon)"></svg>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import feather from 'feather-icons'
import customIcons from '../../utils/icon-helper'

@Component({
  name: 'ZIcon'
})
export default class Icon extends Vue {
  @Prop({ default: '' })
  private icon!: string
  @Prop({default: ''})
  private size!: string

  public getIcon(iconName: string): HTMLOrSVGImageElement {
    let DOM = null
    if (feather.icons[iconName]) {
      DOM = feather.icons[iconName] && feather.icons[iconName].contents
    } else {
      DOM = customIcons[iconName] &&  customIcons[iconName].contents;
    }
    return DOM
  }
}
</script>

<style lang="css" scoped>
  .z-icon {
    @apply  w-6 h-6 fill-current stroke-current text-slate stroke-2;
  }

  .z-icon--small {
    @apply w-4 h-4;
  }
</style>
