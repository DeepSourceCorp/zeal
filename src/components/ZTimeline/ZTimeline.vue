<template>
  <div>
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'ZTimeline',
  props: {},
  data() {
    return {}
  },
  mounted() {
    // this.items = this.$children.filter(child => child.$options.name === 'ZBreadcrumbItem')
    // this.length = this.items.length
    // this.items.map((item, index) => {
    //   item.index = index
    // })
  }
}
</script>
