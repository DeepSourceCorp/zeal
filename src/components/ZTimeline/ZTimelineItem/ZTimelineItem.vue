<template>
  <div class="timeline__item flex relative gap-x-4">
    <div class="flex flex-col justify-center items-center gap-y-2">
      <div class="w-8 h-8 flex justify-center items-center">
        <slot name="icon">
          <span class="inline-block h-full w-full rounded-full bg-juniper"></span>
        </slot>
      </div>
      <span
        class="h-full border-l border-2 rounded-sm left-4"
        :class="{
          'border-transparent': index === lastIndex,
          'border-ink-100': index !== lastIndex
        }"
      ></span>
    </div>
    <slot>
      <div class="flex flex-col space-y-2 text-xs">
        <div class="text-vanilla-100">{{ content }}</div>
        <div class="text-vanilla-400">{{ timestamp }}</div>
      </div>
    </slot>
  </div>
</template>

<script>
export default {
  name: 'ZTimelineItem',
  props: {
    content: {
      type: String,
      default: ''
    },
    timestamp: {
      type: String,
      default: ''
    }
  },
  data() {
    return {
      index: 0
    }
  },
  computed: {
    lastIndex() {
      return this.$parent.length - 1
    }
  }
}
</script>
