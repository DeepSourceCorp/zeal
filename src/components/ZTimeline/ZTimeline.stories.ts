import '../../assets/css/layout.css'
import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'

import ZAvatar from '../ZAvatar/ZAvatar.vue'
import ZIcon from '../ZIcon/ZIcon.vue'
import ZTimeline from './ZTimeline.vue'
import ZTimelineItem from './ZTimelineItem/ZTimelineItem.vue'

export default {
  title: 'Timeline',
  component: ZTimeline,
  excludeStories: /.*Data$/
}

export const BasicTimeline = () => ({
  components: { ZTimeline, ZTimelineItem },
  template: `<div class='padded-container'>
    <z-timeline>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
    </z-timeline>
    </div>`
})

export const TimelineWithCustomIcon = () => ({
  components: { ZTimeline, ZTimelineItem, ZAvatar },
  template: `<div class='padded-container'>
    <z-timeline>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
      ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
    </z-timeline>
    </div>`
})

export const TimelineWithCustomBorderWidth = () => ({
  components: { ZTimeline, ZTimelineItem, ZAvatar },
  template: `<div class='padded-container'>
    <z-timeline>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
    </z-timeline>
    </div>`
})

export const AuditLogEventsAsTimelineItems = () => ({
  components: { ZTimeline, ZTimelineItem, ZIcon },
  data: function () {
    return {
      auditLogs: [
        {
          icon: 'bot',
          descriptionRendered: `<strong>John Doe's brother</strong> updated name of the namespace to <strong>@deepsourcelabs</strong>.`,
          createdAt: '1 month ago'
        },
        {
          icon: 'focus',
          descriptionRendered:
            '<strong>John Doe</strong> updated publish access to the namespace <strong>@deepsource</strong>. Members can now push to <strong>@deepsource/armory</strong>.',
          createdAt: '2 months ago'
        },
        {
          icon: 'bot',
          descriptionRendered: '<strong>John Doe</strong> created the namespace <strong>@deepsource</strong>.',
          createdAt: '3 months ago'
        }
      ]
    }
  },
  template: `<div class='padded-container'>
    <z-timeline>
      <z-timeline-item
      v-for="(log, idx) in auditLogs"
      :key="idx"
      border-width-class=""
      icon-align-items-class="items-baseline"
      class="w-full h-24 md:h-20"
    >
      <template #icon>
        <z-icon :icon="log.icon" />
      </template>

      <div
        class="flex flex-col w-full text-xs md:justify-between gap-y-1 md:gap-y-0 md:flex-row text-vanilla-400"
      >
        <p class="text-xs leading-5 prose" v-html="log.descriptionRendered"></p>

        <span class="leading-5">{{ log.createdAt }}</span>
      </div>
    </z-timeline-item>
    </z-timeline>
    </div>`
})
