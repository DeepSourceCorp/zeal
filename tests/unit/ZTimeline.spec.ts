import ZAvatar from '../../src/components/ZAvatar'
import ZIcon from '../../src/components/ZIcon'
import ZTimeline from '../../src/components/ZTimeline'
import ZTimelineItem from '../../src/components/ZTimeline/ZTimelineItem'

import { mount } from '@vue/test-utils'

const BasicTimeline = {
  template: `<z-timeline>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM"></z-timeline-item>
    </z-timeline>`,
  components: {
    ZTimeline,
    ZTimelineItem
  }
}

const TimelineWithCustomIcon = {
  template: `<z-timeline>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      <z-timeline-item class="h-32" content="Carter Workman" timestamp="Friday, 4:24PM">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar></template>
      </z-timeline-item>
      </z-timeline>`,
  components: {
    ZTimeline,
    ZTimelineItem,
    ZAvatar
  }
}

const TimelineWithCustomBorderWidth = {
  template: `<z-timeline>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        /></template>
      </z-timeline-item>
      <z-timeline-item content="Carter Workman" timestamp="Friday, 4:24PM" border-width-class="" class="h-32">
        <template slot="icon">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
          size="sm"
        /></template>
      </z-timeline-item>
      </z-timeline>`,
  components: {
    ZTimeline,
    ZTimelineItem,
    ZAvatar
  }
}

const AuditLogEventsAsTimelineItems = {
  components: { ZTimeline, ZTimelineItem, ZIcon },
  data: function() {
    return {
      auditLogs: [
        {
          icon: 'bot',
          descriptionRendered: `<strong>John Doe's brother</strong> updated name of the namespace to <strong>@deepsourcelabs</strong>.`,
          createdAt: '1 month ago'
        },
        {
          icon: 'focus',
          descriptionRendered:
            '<strong>John Doe</strong> updated publish access to the namespace <strong>@deepsource</strong>. Members can now push to <strong>@deepsource/armory</strong>.',
          createdAt: '2 months ago'
        },
        {
          icon: 'bot',
          descriptionRendered: '<strong>John Doe</strong> created the namespace <strong>@deepsource</strong>.',
          createdAt: '3 months ago'
        }
      ]
    }
  },
  template: `<div class='padded-container'>
    <z-timeline>
      <z-timeline-item
      v-for="(log, idx) in auditLogs"
      :key="idx"
      border-width-class=""
      icon-align-items-class="items-baseline"
      class="w-full h-24 md:h-20"
    >
      <template #icon>
        <z-icon :icon="log.icon" />
      </template>

      <div
        class="flex flex-col w-full text-xs md:justify-between gap-y-1 md:gap-y-0 md:flex-row text-vanilla-400"
      >
        <p class="text-xs leading-5 prose" v-html="log.descriptionRendered"></p>

        <span class="leading-5">{{ log.createdAt }}</span>
      </div>
    </z-timeline-item>
    </z-timeline>
    </div>`
}

describe('ZTimeline', () => {
  it('renders a default timeline component', () => {
    let wrapper = mount(BasicTimeline)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders a timeline component with custom icon', () => {
    let wrapper = mount(TimelineWithCustomIcon)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders a timeline component with custom border width', () => {
    let wrapper = mount(TimelineWithCustomBorderWidth)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders audit log events as timeline items', () => {
    let wrapper = mount(AuditLogEventsAsTimelineItems)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
