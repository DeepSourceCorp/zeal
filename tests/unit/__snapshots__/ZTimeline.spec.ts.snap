// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZTimeline renders a default timeline component 1`] = `
<div class="timeline flex flex-col gap-y-2">
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><span class="inline-block w-full h-full rounded-full bg-juniper"></span></div> <span class="h-full border-l rounded-sm left-4 border-ink-100 border-2"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><span class="inline-block w-full h-full rounded-full bg-juniper"></span></div> <span class="h-full border-l rounded-sm left-4 border-ink-100 border-2"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
</div>
`;

exports[`ZTimeline renders a timeline component with custom border width 1`] = `
<div class="timeline flex flex-col gap-y-2">
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><a class="inline-block rounded-full bg-ink-300 p-0.5"><img src="https://randomuser.me/api/portraits/women/24.jpg" alt="Akshay Paliwal" class="rounded-full leading-none h-6 w-6 text-xs">
          <!---->
        </a></div> <span class="h-full border-l rounded-sm left-4 border-ink-100"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><a class="inline-block rounded-full bg-ink-300 p-0.5"><img src="https://randomuser.me/api/portraits/women/24.jpg" alt="Akshay Paliwal" class="rounded-full leading-none h-6 w-6 text-xs">
          <!---->
        </a></div> <span class="h-full border-l rounded-sm left-4 border-ink-100"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
</div>
`;

exports[`ZTimeline renders a timeline component with custom icon 1`] = `
<div class="timeline flex flex-col gap-y-2">
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><a class="inline-block rounded-full bg-ink-300 p-0.5"><img src="https://randomuser.me/api/portraits/women/24.jpg" alt="Akshay Paliwal" class="rounded-full leading-none h-6 w-6 text-xs">
          <!---->
        </a></div> <span class="h-full border-l rounded-sm left-4 border-ink-100 border-2"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
  <div class="relative flex timeline__item gap-x-2 h-32">
    <div class="flex flex-col items-center justify-center gap-y-2">
      <div class="flex items-center justify-center w-6 h-6"><a class="inline-block rounded-full bg-ink-300 p-0.5"><img src="https://randomuser.me/api/portraits/women/24.jpg" alt="Akshay Paliwal" class="rounded-full leading-none h-6 w-6 text-xs">
          <!---->
        </a></div> <span class="h-full border-l rounded-sm left-4 border-ink-100 border-2"></span>
    </div>
    <div class="flex flex-col space-y-2 text-xs">
      <div class="text-vanilla-100">Carter Workman</div>
      <div class="text-vanilla-400">Friday, 4:24PM</div>
    </div>
  </div>
</div>
`;
